### ISSUE TITLE
Project Level 5 workaround for test case 4

### ISSUE THREAD

**Issue by vveining** [nus-cs2030/2122-s2#338]
# Summary
Basically, my logic for the project is flawed but it has worked so far up to level 5, resulting in test case 4 of level 5 to have wrong output.

## Description 
Looking for suggestions for a quick workaround, am even okay with hardcode since level 6's code will be vastly different for my case already.

## Screenshots (if any): 
Insert Images here if necessary
![image](https://user-images.githubusercontent.com/94526059/163589634-1ee9acd8-63f4-4894-aa2f-887feb1c86bf.png)

**Comment by nelsonchoo456** [comment]
Hi! I think you are calling supplier.get() too early or at the wrong timings, hence customer 3 and 5 have the wrong service times. Maybe you can try by only calling supplier.get() for the Serve event. 
[Can refer to this issue that was raised previously](https://github.com/nus-cs2030/2122-s2/issues/153)

**Comment by jamietan123** [comment]
I think the easiest way would be to create a new Event before each Serve event, this will help you with level 6 too.

**Comment by ongyongen** [comment]
hi I think im facing a similar issue. Is everything correct (even the statistics printed out match) except this?

**Comment by vveining** [comment]
@ongyongen yeah that stats are correct! i realised the issue but have yet to fix it :( planning to redo my whole code. 

basically the problem occurs when the customer number sequence is different from the serving sequence. i create my done events in order of the customer number and not the time they are served. so for example, cust 1 and 2 are served at 0.700 and 0.600 respectively, so my code results in cust1 being processed first and getting allocated the service time that was supposed to be for cust 2, resulting in both customers having swapped service time.

**Comment by terrrence** [comment]
> I think the easiest way would be to create a new Event before each Serve event, this will help you with level 6 too.

may i ask what does this new event do? im currently stuck on this test case too but im not very sure how i should go about it

**Comment by terrrence** [comment]
hi! not sure if you solved it yet, but i finally solved this test case by only calling supplier.get() in my serve class. 
my current serve event constructor only takes in customer, time, and server.
my problem was that i called supplier.get() in my arrive and wait class. so i think maybe you can try only doing supplier.get() inside the serve event class to get the serve time of the customer!

**Comment by jlee1007** [comment]
think you can only call get in the serve class to avoid such an issue but this may cause problems at l6 which I'm stuck now

**Comment by bbelleeee** [comment]
i think what u cld do is probs create customers with supplier attribute of the dummy lambda ()->1.0, until u run the serve event, then replace it with another version of a serve event with a new customer that takes in the actual supplier from the input!!
